modules/product/src/main/java/com/opengamma/strata/product/bond/ResolvedBillTrade.java

```diff
@@ -0,0 +1,531 @@
+import com.opengamma.strata.collect.ArgChecker;
+import com.opengamma.strata.product.PortfolioItemInfo;
+import com.opengamma.strata.product.ResolvedTrade;
+import com.opengamma.strata.product.TradeInfo;
+
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+
+import org.joda.beans.MetaBean;
+import org.joda.beans.MetaProperty;
+import org.joda.beans.impl.direct.DirectFieldsBeanBuilder;
+import org.joda.beans.impl.direct.DirectMetaBean;
+import org.joda.beans.impl.direct.DirectMetaPropertyMap;
+import org.joda.beans.Bean;
+import org.joda.beans.JodaBeanUtils;
+import org.joda.beans.impl.direct.DirectMetaProperty;
+
+/**
+ * A trade in a bill, resolved for pricing.
+ * <p>
+ * This is the resolved form of {@link BillTrade} and is the primary input to the pricers.
+ * Applications will typically create a {@code ResolvedBillTrade} from a {@code BillTrade}
+ * using {@link BillTrade#resolve(ReferenceData)}.
+ * <p>
+ * A {@code ResolvedBillTrade} is bound to data that changes over time, such as holiday calendars.
+ * If the data changes, such as the addition of a new holiday, the resolved form will not be updated.
+ * Care must be taken when placing the resolved form in a cache or persistence layer.
+ * 
+ * <h4>Price</h4>
+ * Strata uses <i>decimal rates</i> for bills in the trade model, pricers and market data.
+ * For example, a yield of 1.32% is represented in Strata by 0.0132.
+ */
+@BeanDefinition(constructorScope = "package")
+public class ResolvedBillTrade
+    implements ResolvedTrade, ImmutableBean, Serializable {
+
+  /**
+   * The additional information, defaulted to an empty instance.
+   * <p>
+   * This allows additional information to be attached.
+   */
+  @PropertyDefinition(validate = "notNull", overrideGet = true)
+  private final PortfolioItemInfo info;
+  /**
+   * The resolved bill product.
+   * <p>
+   * The product captures the contracted financial details of the trade.
+   */
+  @PropertyDefinition(validate = "notNull", overrideGet = true)
+  private final ResolvedBill product;
+  /**
+   * The quantity, indicating the number of bond contracts in the trade.
+   * <p>
+   * This will be positive if buying and negative if selling.
+   */
+  @PropertyDefinition
+  private final double quantity;
+  /**
+   * The settlement details of the bill trade.
+   * <p>
+   * When this class is used to represent a position, this property will be empty.
+   */
+  @PropertyDefinition(get = "optional")
+  private final Payment settlement;
+
+  //-------------------------------------------------------------------------
+  @ImmutableDefaults
+  private static void applyDefaults(Builder builder) {
+    builder.info = TradeInfo.empty();
+  }
+
+  @ImmutablePreBuild
+  private static void preBuild(Builder builder) {
+    if (builder.settlement != null) {
+      ArgChecker.isTrue(builder.quantity * builder.settlement.getAmount() <= 0); // Amount negative for buy
```

Comment:
Is this right?

Reply:
When buying, quantity > 0 and amount is paid and so < 0. This is to check that the user does not get confused on the signs.